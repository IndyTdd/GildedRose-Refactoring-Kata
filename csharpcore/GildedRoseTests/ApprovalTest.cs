using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Combinations;
using Xunit;
using ApprovalTests.Reporters;
using GildedRoseKata;

namespace GildedRoseTests
{
    [UseReporter(typeof(DiffReporter))]
    public class ApprovalTest
    {
        [Fact]
        public void Verify()
        {
            var names = new List<string>
            {
                "Aged Brie",
                "Backstage passes to a TAFKAL80ETC concert",
                "Sulfuras, Hand of Ragnaros",
                "Generic1",
            };
            var qualities = new List<int>
            {
                0,
                1,
                -1,
                50,
                49,
                80
            };

            var sellIns = new List<int>
            {
                0,
                6,
                5,
                -1,
                10,
                11,
                12,
                30,
            };
            CombinationApprovals.VerifyAllCombinations(DoUpdateQuality, names, qualities, sellIns);
        }
        
        private string DoUpdateQuality(string name, int quality, int sellIn)
        {
            var item = new Item() {Name = name, Quality = quality, SellIn = sellIn};
            var items = new List<Item>()
            {
                item,
            };
            var app = new GildedRose(items);
                
            app.UpdateQuality();

            return item.ToString();
        }
    }
}
