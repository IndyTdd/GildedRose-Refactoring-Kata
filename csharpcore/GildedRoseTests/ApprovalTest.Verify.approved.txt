[Aged Brie,0,0] => Item { Name: Aged Brie, SellIn: -1, Quality: 2 }
[Aged Brie,0,6] => Item { Name: Aged Brie, SellIn: 5, Quality: 1 }
[Aged Brie,0,5] => Item { Name: Aged Brie, SellIn: 4, Quality: 1 }
[Aged Brie,0,-1] => Item { Name: Aged Brie, SellIn: -2, Quality: 2 }
[Aged Brie,0,10] => Item { Name: Aged Brie, SellIn: 9, Quality: 1 }
[Aged Brie,0,11] => Item { Name: Aged Brie, SellIn: 10, Quality: 1 }
[Aged Brie,0,12] => Item { Name: Aged Brie, SellIn: 11, Quality: 1 }
[Aged Brie,0,30] => Item { Name: Aged Brie, SellIn: 29, Quality: 1 }
[Aged Brie,1,0] => Item { Name: Aged Brie, SellIn: -1, Quality: 3 }
[Aged Brie,1,6] => Item { Name: Aged Brie, SellIn: 5, Quality: 2 }
[Aged Brie,1,5] => Item { Name: Aged Brie, SellIn: 4, Quality: 2 }
[Aged Brie,1,-1] => Item { Name: Aged Brie, SellIn: -2, Quality: 3 }
[Aged Brie,1,10] => Item { Name: Aged Brie, SellIn: 9, Quality: 2 }
[Aged Brie,1,11] => Item { Name: Aged Brie, SellIn: 10, Quality: 2 }
[Aged Brie,1,12] => Item { Name: Aged Brie, SellIn: 11, Quality: 2 }
[Aged Brie,1,30] => Item { Name: Aged Brie, SellIn: 29, Quality: 2 }
[Aged Brie,-1,0] => Item { Name: Aged Brie, SellIn: -1, Quality: 1 }
[Aged Brie,-1,6] => Item { Name: Aged Brie, SellIn: 5, Quality: 0 }
[Aged Brie,-1,5] => Item { Name: Aged Brie, SellIn: 4, Quality: 0 }
[Aged Brie,-1,-1] => Item { Name: Aged Brie, SellIn: -2, Quality: 1 }
[Aged Brie,-1,10] => Item { Name: Aged Brie, SellIn: 9, Quality: 0 }
[Aged Brie,-1,11] => Item { Name: Aged Brie, SellIn: 10, Quality: 0 }
[Aged Brie,-1,12] => Item { Name: Aged Brie, SellIn: 11, Quality: 0 }
[Aged Brie,-1,30] => Item { Name: Aged Brie, SellIn: 29, Quality: 0 }
[Aged Brie,50,0] => Item { Name: Aged Brie, SellIn: -1, Quality: 50 }
[Aged Brie,50,6] => Item { Name: Aged Brie, SellIn: 5, Quality: 50 }
[Aged Brie,50,5] => Item { Name: Aged Brie, SellIn: 4, Quality: 50 }
[Aged Brie,50,-1] => Item { Name: Aged Brie, SellIn: -2, Quality: 50 }
[Aged Brie,50,10] => Item { Name: Aged Brie, SellIn: 9, Quality: 50 }
[Aged Brie,50,11] => Item { Name: Aged Brie, SellIn: 10, Quality: 50 }
[Aged Brie,50,12] => Item { Name: Aged Brie, SellIn: 11, Quality: 50 }
[Aged Brie,50,30] => Item { Name: Aged Brie, SellIn: 29, Quality: 50 }
[Aged Brie,49,0] => Item { Name: Aged Brie, SellIn: -1, Quality: 50 }
[Aged Brie,49,6] => Item { Name: Aged Brie, SellIn: 5, Quality: 50 }
[Aged Brie,49,5] => Item { Name: Aged Brie, SellIn: 4, Quality: 50 }
[Aged Brie,49,-1] => Item { Name: Aged Brie, SellIn: -2, Quality: 50 }
[Aged Brie,49,10] => Item { Name: Aged Brie, SellIn: 9, Quality: 50 }
[Aged Brie,49,11] => Item { Name: Aged Brie, SellIn: 10, Quality: 50 }
[Aged Brie,49,12] => Item { Name: Aged Brie, SellIn: 11, Quality: 50 }
[Aged Brie,49,30] => Item { Name: Aged Brie, SellIn: 29, Quality: 50 }
[Aged Brie,80,0] => Item { Name: Aged Brie, SellIn: -1, Quality: 80 }
[Aged Brie,80,6] => Item { Name: Aged Brie, SellIn: 5, Quality: 80 }
[Aged Brie,80,5] => Item { Name: Aged Brie, SellIn: 4, Quality: 80 }
[Aged Brie,80,-1] => Item { Name: Aged Brie, SellIn: -2, Quality: 80 }
[Aged Brie,80,10] => Item { Name: Aged Brie, SellIn: 9, Quality: 80 }
[Aged Brie,80,11] => Item { Name: Aged Brie, SellIn: 10, Quality: 80 }
[Aged Brie,80,12] => Item { Name: Aged Brie, SellIn: 11, Quality: 80 }
[Aged Brie,80,30] => Item { Name: Aged Brie, SellIn: 29, Quality: 80 }
[Backstage passes to a TAFKAL80ETC concert,0,0] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,0,6] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 5, Quality: 2 }
[Backstage passes to a TAFKAL80ETC concert,0,5] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 3 }
[Backstage passes to a TAFKAL80ETC concert,0,-1] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,0,10] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 2 }
[Backstage passes to a TAFKAL80ETC concert,0,11] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 1 }
[Backstage passes to a TAFKAL80ETC concert,0,12] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 11, Quality: 1 }
[Backstage passes to a TAFKAL80ETC concert,0,30] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 29, Quality: 1 }
[Backstage passes to a TAFKAL80ETC concert,1,0] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,1,6] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 5, Quality: 3 }
[Backstage passes to a TAFKAL80ETC concert,1,5] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 4 }
[Backstage passes to a TAFKAL80ETC concert,1,-1] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,1,10] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 3 }
[Backstage passes to a TAFKAL80ETC concert,1,11] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 2 }
[Backstage passes to a TAFKAL80ETC concert,1,12] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 11, Quality: 2 }
[Backstage passes to a TAFKAL80ETC concert,1,30] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 29, Quality: 2 }
[Backstage passes to a TAFKAL80ETC concert,-1,0] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,-1,6] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 5, Quality: 1 }
[Backstage passes to a TAFKAL80ETC concert,-1,5] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 2 }
[Backstage passes to a TAFKAL80ETC concert,-1,-1] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,-1,10] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 1 }
[Backstage passes to a TAFKAL80ETC concert,-1,11] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,-1,12] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 11, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,-1,30] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 29, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,50,0] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,50,6] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 5, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,50,5] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,50,-1] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,50,10] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,50,11] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,50,12] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 11, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,50,30] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 29, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,49,0] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,49,6] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 5, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,49,5] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,49,-1] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,49,10] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,49,11] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,49,12] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 11, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,49,30] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 29, Quality: 50 }
[Backstage passes to a TAFKAL80ETC concert,80,0] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,80,6] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 5, Quality: 80 }
[Backstage passes to a TAFKAL80ETC concert,80,5] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 80 }
[Backstage passes to a TAFKAL80ETC concert,80,-1] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0 }
[Backstage passes to a TAFKAL80ETC concert,80,10] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 80 }
[Backstage passes to a TAFKAL80ETC concert,80,11] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 10, Quality: 80 }
[Backstage passes to a TAFKAL80ETC concert,80,12] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 11, Quality: 80 }
[Backstage passes to a TAFKAL80ETC concert,80,30] => Item { Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 29, Quality: 80 }
[Sulfuras, Hand of Ragnaros,0,0] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 0 }
[Sulfuras, Hand of Ragnaros,0,6] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 6, Quality: 0 }
[Sulfuras, Hand of Ragnaros,0,5] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 0 }
[Sulfuras, Hand of Ragnaros,0,-1] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 0 }
[Sulfuras, Hand of Ragnaros,0,10] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 0 }
[Sulfuras, Hand of Ragnaros,0,11] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: 0 }
[Sulfuras, Hand of Ragnaros,0,12] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 12, Quality: 0 }
[Sulfuras, Hand of Ragnaros,0,30] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 30, Quality: 0 }
[Sulfuras, Hand of Ragnaros,1,0] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 1 }
[Sulfuras, Hand of Ragnaros,1,6] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 6, Quality: 1 }
[Sulfuras, Hand of Ragnaros,1,5] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 1 }
[Sulfuras, Hand of Ragnaros,1,-1] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 1 }
[Sulfuras, Hand of Ragnaros,1,10] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 1 }
[Sulfuras, Hand of Ragnaros,1,11] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: 1 }
[Sulfuras, Hand of Ragnaros,1,12] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 12, Quality: 1 }
[Sulfuras, Hand of Ragnaros,1,30] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 30, Quality: 1 }
[Sulfuras, Hand of Ragnaros,-1,0] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: -1 }
[Sulfuras, Hand of Ragnaros,-1,6] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 6, Quality: -1 }
[Sulfuras, Hand of Ragnaros,-1,5] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: -1 }
[Sulfuras, Hand of Ragnaros,-1,-1] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: -1 }
[Sulfuras, Hand of Ragnaros,-1,10] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: -1 }
[Sulfuras, Hand of Ragnaros,-1,11] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: -1 }
[Sulfuras, Hand of Ragnaros,-1,12] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 12, Quality: -1 }
[Sulfuras, Hand of Ragnaros,-1,30] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 30, Quality: -1 }
[Sulfuras, Hand of Ragnaros,50,0] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 50 }
[Sulfuras, Hand of Ragnaros,50,6] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 6, Quality: 50 }
[Sulfuras, Hand of Ragnaros,50,5] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 50 }
[Sulfuras, Hand of Ragnaros,50,-1] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 50 }
[Sulfuras, Hand of Ragnaros,50,10] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 50 }
[Sulfuras, Hand of Ragnaros,50,11] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: 50 }
[Sulfuras, Hand of Ragnaros,50,12] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 12, Quality: 50 }
[Sulfuras, Hand of Ragnaros,50,30] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 30, Quality: 50 }
[Sulfuras, Hand of Ragnaros,49,0] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 49 }
[Sulfuras, Hand of Ragnaros,49,6] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 6, Quality: 49 }
[Sulfuras, Hand of Ragnaros,49,5] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 49 }
[Sulfuras, Hand of Ragnaros,49,-1] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 49 }
[Sulfuras, Hand of Ragnaros,49,10] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 49 }
[Sulfuras, Hand of Ragnaros,49,11] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: 49 }
[Sulfuras, Hand of Ragnaros,49,12] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 12, Quality: 49 }
[Sulfuras, Hand of Ragnaros,49,30] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 30, Quality: 49 }
[Sulfuras, Hand of Ragnaros,80,0] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 80 }
[Sulfuras, Hand of Ragnaros,80,6] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 6, Quality: 80 }
[Sulfuras, Hand of Ragnaros,80,5] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 80 }
[Sulfuras, Hand of Ragnaros,80,-1] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 80 }
[Sulfuras, Hand of Ragnaros,80,10] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 80 }
[Sulfuras, Hand of Ragnaros,80,11] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 11, Quality: 80 }
[Sulfuras, Hand of Ragnaros,80,12] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 12, Quality: 80 }
[Sulfuras, Hand of Ragnaros,80,30] => Item { Name: Sulfuras, Hand of Ragnaros, SellIn: 30, Quality: 80 }
[Generic1,0,0] => Item { Name: Generic1, SellIn: -1, Quality: 0 }
[Generic1,0,6] => Item { Name: Generic1, SellIn: 5, Quality: 0 }
[Generic1,0,5] => Item { Name: Generic1, SellIn: 4, Quality: 0 }
[Generic1,0,-1] => Item { Name: Generic1, SellIn: -2, Quality: 0 }
[Generic1,0,10] => Item { Name: Generic1, SellIn: 9, Quality: 0 }
[Generic1,0,11] => Item { Name: Generic1, SellIn: 10, Quality: 0 }
[Generic1,0,12] => Item { Name: Generic1, SellIn: 11, Quality: 0 }
[Generic1,0,30] => Item { Name: Generic1, SellIn: 29, Quality: 0 }
[Generic1,1,0] => Item { Name: Generic1, SellIn: -1, Quality: 0 }
[Generic1,1,6] => Item { Name: Generic1, SellIn: 5, Quality: 0 }
[Generic1,1,5] => Item { Name: Generic1, SellIn: 4, Quality: 0 }
[Generic1,1,-1] => Item { Name: Generic1, SellIn: -2, Quality: 0 }
[Generic1,1,10] => Item { Name: Generic1, SellIn: 9, Quality: 0 }
[Generic1,1,11] => Item { Name: Generic1, SellIn: 10, Quality: 0 }
[Generic1,1,12] => Item { Name: Generic1, SellIn: 11, Quality: 0 }
[Generic1,1,30] => Item { Name: Generic1, SellIn: 29, Quality: 0 }
[Generic1,-1,0] => Item { Name: Generic1, SellIn: -1, Quality: -1 }
[Generic1,-1,6] => Item { Name: Generic1, SellIn: 5, Quality: -1 }
[Generic1,-1,5] => Item { Name: Generic1, SellIn: 4, Quality: -1 }
[Generic1,-1,-1] => Item { Name: Generic1, SellIn: -2, Quality: -1 }
[Generic1,-1,10] => Item { Name: Generic1, SellIn: 9, Quality: -1 }
[Generic1,-1,11] => Item { Name: Generic1, SellIn: 10, Quality: -1 }
[Generic1,-1,12] => Item { Name: Generic1, SellIn: 11, Quality: -1 }
[Generic1,-1,30] => Item { Name: Generic1, SellIn: 29, Quality: -1 }
[Generic1,50,0] => Item { Name: Generic1, SellIn: -1, Quality: 48 }
[Generic1,50,6] => Item { Name: Generic1, SellIn: 5, Quality: 49 }
[Generic1,50,5] => Item { Name: Generic1, SellIn: 4, Quality: 49 }
[Generic1,50,-1] => Item { Name: Generic1, SellIn: -2, Quality: 48 }
[Generic1,50,10] => Item { Name: Generic1, SellIn: 9, Quality: 49 }
[Generic1,50,11] => Item { Name: Generic1, SellIn: 10, Quality: 49 }
[Generic1,50,12] => Item { Name: Generic1, SellIn: 11, Quality: 49 }
[Generic1,50,30] => Item { Name: Generic1, SellIn: 29, Quality: 49 }
[Generic1,49,0] => Item { Name: Generic1, SellIn: -1, Quality: 47 }
[Generic1,49,6] => Item { Name: Generic1, SellIn: 5, Quality: 48 }
[Generic1,49,5] => Item { Name: Generic1, SellIn: 4, Quality: 48 }
[Generic1,49,-1] => Item { Name: Generic1, SellIn: -2, Quality: 47 }
[Generic1,49,10] => Item { Name: Generic1, SellIn: 9, Quality: 48 }
[Generic1,49,11] => Item { Name: Generic1, SellIn: 10, Quality: 48 }
[Generic1,49,12] => Item { Name: Generic1, SellIn: 11, Quality: 48 }
[Generic1,49,30] => Item { Name: Generic1, SellIn: 29, Quality: 48 }
[Generic1,80,0] => Item { Name: Generic1, SellIn: -1, Quality: 78 }
[Generic1,80,6] => Item { Name: Generic1, SellIn: 5, Quality: 79 }
[Generic1,80,5] => Item { Name: Generic1, SellIn: 4, Quality: 79 }
[Generic1,80,-1] => Item { Name: Generic1, SellIn: -2, Quality: 78 }
[Generic1,80,10] => Item { Name: Generic1, SellIn: 9, Quality: 79 }
[Generic1,80,11] => Item { Name: Generic1, SellIn: 10, Quality: 79 }
[Generic1,80,12] => Item { Name: Generic1, SellIn: 11, Quality: 79 }
[Generic1,80,30] => Item { Name: Generic1, SellIn: 29, Quality: 79 }
